hazelcast/src/main/java/com/hazelcast/datastore/ExternalDataStoreFactory.java

```diff
@@ -26,17 +26,27 @@
  * @since 5.2
  */
 @Beta
-public interface ExternalDataStoreFactory<DS> {
+public interface ExternalDataStoreFactory<DS> extends AutoCloseable {
     /**
      * Returns configured data store. Depending on configuration and implementation it can create new data store
      * or reuse existing one.
      */
-    DS getDataStore();
+    DataStoreHolder<DS> createDataStore();
```

Comment:
Since this PR targets 5.2.1, it can be good to think about api compatibility. `ExternalDataStoreFactory` is beta api but seems to me it is not clear which changes can be done on beta-apis between versions. 

Changing it in a patch, doesn't feel good, it will not be a drop-in replacement any more.